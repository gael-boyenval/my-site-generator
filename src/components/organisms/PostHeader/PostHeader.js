import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { unit, colors } from 'Styles/tokens';
import { rem } from 'Styles/utils';
import { Margin } from 'Layout/Spaces';
import Grid from 'Layout/Grid';
import Container from 'Layout/Container';

import { EditorialTitle, LeadBodyCopyEditorial, TinyHeading } from 'Atoms/Typography';

const Header = styled.div`
  background: ${colors.white.light};
  color: ${colors.base.darkest};
  min-height: ${rem(unit * 20)};
  margin-top: -${rem(unit * 20)};
  padding-top: ${rem(unit * 6)};
`;

const ImageWrapper = styled.div`
  background: ${colors.base.darkest};
  min-height: ${rem(unit * 35)};
`;

const Wrapper = styled.header`
  background: ${colors.white.light};
`;

const PostHeader = ({ title, publishDate, leadBodyCopy }) => (
  <Wrapper>
    <ImageWrapper />
    <Container>
      <Grid gutter>
        <Grid.Cell size={9 / 12}>
          <Header>
            <Grid gutter>
              <Grid.Cell size={7 / 9} push={1 / 9}>
                <TinyHeading>{publishDate}</TinyHeading>
                <Margin vertical="m">
                  <EditorialTitle>{title}</EditorialTitle>
                </Margin>
                <LeadBodyCopyEditorial>{leadBodyCopy}</LeadBodyCopyEditorial>
              </Grid.Cell>
            </Grid>
          </Header>
        </Grid.Cell>
      </Grid>
    </Container>
  </Wrapper>
);

PostHeader.propTypes = {
  title: PropTypes.string.isRequired,
  publishDate: PropTypes.string.isRequired,
  leadBodyCopy: PropTypes.string.isRequired,
};

export default PostHeader;
