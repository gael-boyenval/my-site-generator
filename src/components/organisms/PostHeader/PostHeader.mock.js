export default {
  title: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod.',
  leadBodyCopy:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  publishDate: '2018-04-09',
};
