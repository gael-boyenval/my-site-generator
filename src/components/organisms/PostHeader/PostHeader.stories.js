/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

import PostHeader from './PostHeader';
import propMock from './PostHeader.mock';

storiesOf('04 - Organisms', module)
  .addDecorator(withKnobs)
  .add('PostHeader', withInfo()(() => <PostHeader {...propMock} />));
