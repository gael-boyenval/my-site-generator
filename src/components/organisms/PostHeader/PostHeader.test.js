/* eslint-disable no-undef */
import React from 'react';

import PostHeader from './PostHeader';
import propMock from './PostHeader.mock';

describe('Component PostHeader', () => {
  it('should be render without crash', () => {
    const component = shallow(<PostHeader {...propMock} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
