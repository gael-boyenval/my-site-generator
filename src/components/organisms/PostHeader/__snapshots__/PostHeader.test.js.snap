// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PostHeader should be render without crash 1`] = `
<PostHeader__Wrapper>
  <PostHeader__ImageWrapper />
  <Container>
    <Grid
      gutter={true}
    >
      <Grid__GridCell
        push={null}
        size={0.75}
      >
        <PostHeader__Header>
          <Grid
            gutter={true}
          >
            <Grid__GridCell
              push={0.1111111111111111}
              size={0.7777777777777778}
            >
              <Typography__TinyHeading>
                2018-04-09
              </Typography__TinyHeading>
              <Spaces__Margin
                all={false}
                bottom={false}
                horizontal={false}
                inline={false}
                left={false}
                right={false}
                top={false}
                vertical="m"
              >
                <Typography__EditorialTitle>
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod.
                </Typography__EditorialTitle>
              </Spaces__Margin>
              <Typography__LeadBodyCopyEditorial>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
              </Typography__LeadBodyCopyEditorial>
            </Grid__GridCell>
          </Grid>
        </PostHeader__Header>
      </Grid__GridCell>
    </Grid>
  </Container>
</PostHeader__Wrapper>
`;
